pkg/controller/perconaservermongodb/backup.go

```diff
@@ -364,11 +367,115 @@ func secretExists(ctx context.Context, cl client.Client, nn types.NamespacedName
 	var secret corev1.Secret
 	err := cl.Get(ctx, nn, &secret)
 	if err != nil {
-		if k8serrors.IsNotFound(err) {
+		if k8sErrors.IsNotFound(err) {
 			return false, nil
 		}
 		return false, err
 	}
 
 	return true, nil
 }
+
+func isPodUpToDate(pod *corev1.Pod, stsRevision, image string) bool {
+	if pod.Labels["controller-revision-hash"] != stsRevision {
+		return false
+	}
+
+	for _, ct := range pod.Spec.Containers {
+		if ct.Name == naming.ContainerBackupAgent {
+			if ct.Image != image {
+				return false
+			}
+		}
+	}
+
+	return true
+}
+
+func (r *ReconcilePerconaServerMongoDB) reconcileBackupVersion(ctx context.Context, cr *api.PerconaServerMongoDB) error {
+	log := logf.FromContext(ctx)
+
+	if !cr.Spec.Backup.Enabled {
+		return nil
+	}
+
+	if cr.Status.State != api.AppStateReady {
+		return nil
+	}
+
+	if cr.Status.BackupVersion != "" && cr.Status.BackupImage == cr.Spec.Backup.Image {
+		return nil
+	}
+
+	var rs *api.ReplsetSpec
+	for _, r := range cr.Spec.Replsets {
+		rs = r
+		break
+	}
+
+	stsName := naming.MongodStatefulSetName(cr, rs)
+	sts := psmdb.NewStatefulSet(stsName, cr.Namespace)
+	err := r.client.Get(ctx, client.ObjectKeyFromObject(sts), sts)
+	if err != nil {
+		return errors.Wrapf(err, "get statefulset/%s", stsName)
+	}
+
+	matchLabels := naming.RSLabels(cr, rs)
+	label, ok := sts.Labels[naming.LabelKubernetesComponent]
+	if ok {
+		matchLabels[naming.LabelKubernetesComponent] = label
+	}
```

Comment:
Since we know that we want the `mongod` component, why are we getting the sts just for calculating the label. Maybe we can skip this API call and just ask for the pods with the expected component.

Reply:
it's ok to directly use mongod component but we are getting sts to check if pod is up to date, not to get to component. so we won't be skipping the api call